File: C++/Leetcode/Array/removeDuplicatesII.cpp
#include <iostream>
#include <vector>

class Solution {
public:
    int removeDuplicates(std::vector<int>& nums){
        int l = 1, count = 1;

        for (int r = 1; r < nums.size(); ++r){
            if (nums[r - 1] == nums[r] && count < 2){
                nums[l++] = nums[r];
                count++;
            }
            else if (nums[r - 1] != nums[r]){
                nums[l++] = nums[r];
                count = 1;
            }
        }
        printVector(nums);
        return l;
    }

private:
    void printVector(const std::vector<int>& nums){
        for (const auto& num : nums){
            std::cout << num << std::endl;
        }
    }
};

int main(){
    Solution solution_instance;
    std::vector<int> nums = {0,0,1,1,1,1,2,3,3}; 
    std::cout << solution_instance.removeDuplicates(nums) << std::endl; // Prints 7
}

File: Python/Leetcode/Array/removeDuplicatesII.py
class Solution:
    def removeDuplicates(self, nums):
        l = 1
        count = 1
        
        if len(nums) < 2:
            return l
        
        for r in range(1, len(nums)):
            if nums[r - 1] == nums[r] and count < 2:
                count += 1
                nums[l] = nums[r]
                l += 1
            elif nums[r - 1] == nums[r]:
                count += 1
            elif nums[r - 1] != nums[r]:
                count = 1
                nums[l] = nums[r]
                l += 1   
        nums[l-1] = nums[r]  
        
        print(nums, l)           
        return l
            
                
    
if __name__ == '__main__':
    solution_instance = Solution()
    solution_instance.removeDuplicates([1, 2, 2, 2, 3])